flagsmith/models.go

```diff
@@ -85,14 +85,155 @@ func (f *FlagResourceData) ToClientFS(featureStateID int64, feature int64, envir
 	}
 }
 
-// Generate a new FlagResourceData from client `FeatureState`
-func MakeFlagResourceDataFromClientFS(clientFS *flagsmithapi.FeatureState) FlagResourceData {
+// Generate a new FeatureStateResourceData from client `FeatureState`
+func MakeFeatureStateResourceDataFromClientFS(clientFS *flagsmithapi.FeatureState) FeatureStateResourceData {
 	fsValue := MakeFeatureStateValueFromClientFSV(clientFS.FeatureStateValue)
-	return FlagResourceData{
+	return FeatureStateResourceData{
 		ID:                types.Number{Value: big.NewFloat(float64(clientFS.ID))},
 		Enabled:           types.Bool{Value: clientFS.Enabled},
 		FeatureStateValue: &fsValue,
 		Feature:           types.Number{Value: big.NewFloat(float64(clientFS.Feature))},
 		Environment:       types.Number{Value: big.NewFloat(float64(clientFS.Environment))},
 	}
 }
+
+type MultivariateOption struct {
+	Type                        types.String `tfsdk:"type"`
+	ID                          types.Number `tfsdk:"id"`
+	IntegerValue                types.Number `tfsdk:"integer_value"`
+	StringValue                 types.String `tfsdk:"string_value"`
+	BooleanValue                types.Bool   `tfsdk:"boolean_value"`
+	DefaultPercentageAllocation types.Number `tfsdk:"default_percentage_allocation"`
+}
+
+func (m *MultivariateOption) ToClientMultivariateOption() *flagsmithapi.MultivariateOption {
+	defaultPercentageAllocation, _ := m.DefaultPercentageAllocation.Value.Float64()
+	stringValue := m.StringValue.Value
+	booleanValue := m.BooleanValue.Value
+
+	mo := flagsmithapi.MultivariateOption{
+		Type:                        m.Type.Value,
+		DefaultPercentageAllocation: defaultPercentageAllocation,
+	}
+	if m.ID.Value != nil {
+		moID, _ := m.ID.Value.Int64()
+		mo.ID = &moID
+	}
+	if m.IntegerValue.Value != nil {
```

Comment:
Is there an issue with setting the attributes on the `flagsmithapi.MultivariateOption` model to `nil`? Why do we need to only set the attributes if they're not null?

Reply:
No, they are being set to nil (by not being set explicitly), `if m.IntegerValue.Value != nil` this check here is on the state  `*MultivariateOption` and before we can dereference the pointer we need to make sure it's not nil

Reply:
Ah yes, ok, pointers and stuff 😄 . Gotcha!